File: spec/java/src/io/kaitai/struct/spec/TestSwitchManualIntSizeElse.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.SwitchManualIntSizeElse;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestSwitchManualIntSizeElse extends CommonSpec {
    @Test
    public void testSwitchManualIntSizeElse() throws Exception {
        SwitchManualIntSizeElse r = SwitchManualIntSizeElse.fromFile(SRC_DIR + "switch_tlv.bin");

        assertEquals(r.chunks().size(), 4);

        assertEquals(r.chunks().get(0).code(), 0x11);
        SwitchManualIntSizeElse.Chunk.ChunkMeta meta = (SwitchManualIntSizeElse.Chunk.ChunkMeta) r.chunks().get(0).body();
        assertEquals(meta.title(), "Stuff");
        assertEquals(meta.author(), "Me");

        assertEquals(r.chunks().get(1).code(), 0x22);
        assertEquals(
                ((SwitchManualIntSizeElse.Chunk.ChunkDir) r.chunks().get(1).body()).entries().toArray(),
                new String[] { "AAAA", "BBBB", "CCCC" }
        );

        assertEquals(r.chunks().get(2).code(), 0x33);
        assertEquals(
                ((SwitchManualIntSizeElse.Chunk.Dummy) r.chunks().get(2).body()).rest(),
                new byte[] { 0x10, 0x20, 0x30, 0x40, 0x50, 0x60, 0x70, (byte) 0x80 }
        );

        assertEquals(r.chunks().get(3).code(), 0xff);
        assertEquals(
                ((SwitchManualIntSizeElse.Chunk.Dummy) r.chunks().get(3).body()).rest(),
                new byte[] {}
        );
    }
}

File: spec/python/test_switch_manual_int_size_else.py
import unittest

from switch_manual_int_size_else import SwitchManualIntSizeElse

class TestSwitchManualIntSizeElse(unittest.TestCase):
    def test_switch_manual_int_size_else(self):
        r = SwitchManualIntSizeElse.from_file("src/switch_tlv.bin")

        self.assertEquals(len(r.chunks), 4)

        self.assertEquals(r.chunks[0].code, 0x11)
        meta = r.chunks[0].body
        self.assertEquals(meta.title, "Stuff")
        self.assertEquals(meta.author, "Me")

        self.assertEquals(r.chunks[1].code, 0x22)
        self.assertEquals(r.chunks[1].body.entries, ["AAAA", "BBBB", "CCCC"])

        self.assertEquals(r.chunks[2].code, 0x33)
        self.assertEquals(r.chunks[2].body.rest, bytearray([0x10, 0x20, 0x30, 0x40, 0x50, 0x60, 0x70, 0x80]))

        self.assertEquals(r.chunks[3].code, 0xff)
        self.assertEquals(r.chunks[3].body.rest, "")
